src/viam/examples/modules/complex/test_complex_module.cpp

```diff
@@ -22,6 +22,22 @@
 #include "summation/impl.hpp"
 
 using namespace viam::sdk;
+using namespace viam::sdktests;
+
+struct RegisterGizmoAndSummationFixture {
+    RegisterGizmoAndSummationFixture() {
+        Registry::register_resource(Gizmo::static_api(), Gizmo::resource_registration());
+        Registry::register_resource(Summation::static_api(), Summation::resource_registration());
+    }
+
+    // Test teardown is a noop;
+    ~RegisterGizmoAndSummationFixture() = default;
+};
+
+// Gizmo and Summation APIs must be registered before these tests are run
+// (otherwise client_to_mock_pipeline will not be able to use their resource
+// registrations to create the appropriate servers).
+BOOST_GLOBAL_FIXTURE(RegisterGizmoAndSummationFixture);
```

Comment:
See my comment above. Because `client_to_mock_pipeline` now uses the `mock`'s API's resource registration to call `create_resource_server`, we need to ensure `Gizmo` and `Summation` APIs are actually registered before the following tests run. [Here](https://www.boost.org/doc/libs/1_36_0/libs/test/doc/html/utf/user-guide/fixture/global.html) is more information on boost global fixtures if you're curious.

Reply:
Ah great point; didn't see that deprecation warning. Switched to `BOOST_TEST_GLOBAL_FIXTURE`.